lib/url.c

```diff
@@ -590,6 +590,9 @@ void Curl_conn_free(struct Curl_easy *data, struct connectdata *conn)
 #endif
   Curl_safefree(conn->destination);
 
+  if(conn->handler && conn->handler->disconnect &&
```

Comment:
I think running this before discarding the connection filters is better. So, as a first step. Some `disconnect` handlers do more than freeing memory.